src/App.tsx

```diff
@@ -1,26 +1,113 @@
-import React from 'react';
-import logo from './logo.svg';
-import './App.css';
+import React, { useState } from 'react';
+import { TodoForma } from './components/TodoForma';
+import { TodoList } from './components/TodoList';
+import { IMessage_Date, ITodo } from './interfaces';
+import { MessageInput } from './components/MessageInput';
+import { MessageOutput } from './components/MessageOutput';
+import { ContextMenu } from './components/ContextMenu';
+
+const App: React.FC = () => {
+
+  const [todos, setTodos] = useState<ITodo[]>([]);
+
+  let chatName_exist:boolean = false;
+
+  const addHandler = (title:string) => {
+    const newTodo: ITodo = {
+      title,
+      id: Date.now(),
+      completed: false,
+      container: [],
+      showContext: false,
+      x:"0px",
+      y:"0px",
+
+    };
+    todos.map((todo) => {
+      if (todo.title === title) {
+        chatName_exist = true;
+        alert('Chat with same name already exist!!!');
+      }
+      return todos;
+    });
+
+    if (!chatName_exist) {
+      setTodos((prev) => [newTodo, ...prev]);
+    }
+    chatName_exist = false;
+  };
+
+  const toggleHandler = (id: number) => {
+    setTodos((prev) => prev.map((todo) => {
+      if (todo.id === id) {
+        return {
+          ...todo,
+          completed: !todo.completed,
+        };
+      } return { ...todo, completed: false };
+    }));
+  };
+
+  const removeHandler = (id: number) => {
+    setTodos((prev) => prev.filter((todo) => todo.id !== id));
+  };
+
+  const addMessageHandler = (msg:IMessage_Date) => {
+    setTodos(
+      todos.map((todo) => {
+        if (todo.completed) { todo.container.push(msg); }
```

Comment:
we are talking about chat and messages here. Why do we use `todo` and `completed` names here. Hard to tell what is it.